src/Checkout/AddressForm.js

```diff
@@ -0,0 +1,332 @@
+          });
+          const response = await axios.post(
+            `https://api-prestore.prerananawar1.repl.co/user-details/address/${currentUser._id}/${editAddID}`,
+            {
+              updateAddress: { ...formState },
+            }
+          );
+          if (response.status === 200) {
+            userDispatch({
+              type: "EDIT_ADDRESS",
+              payload: { id: editAddID, address: response.data.address },
+            });
+            console.log("Hello");
+            console.log({ addNewAddress });
+            setMsg("");
+            setAddNewAddress(false);
+            formDispatch({
+              type: "RESET_FORM",
+              payload: initialFormState,
+            });
+          }
+        } catch (error) {
+          userDispatch({
+            type: "STATUS",
+            payload: "Couldn't update address..",
+          });
+        } finally {
+          userDispatch({ type: "STATUS", payload: "" });
+        }
+      } else {
+        setMsg("Please enter a valid Indian Zip-Code");
+      }
+    } else {
+      setMsg("Please enter a valid 10 digit phone number");
+    }
+  };
+
+  return (
+    <div>
+      <form
+        onSubmit={editAddID ? handleEditAddress : handleSubmit}
+        className='address-form'
+      >
+        <div className='flex-input'>
+          <div className='input'>
+            <input
+              type='text'
+              className='input-txt'
+              required
+              value={formState.name}
+              onChange={(e) =>
+                formDispatch({ type: "SET_NAME", payload: e.target.value })
+              }
+            />
+            <span className='flt-label'>Your Name</span>
+          </div>
+
+          <div className='input'>
+            <input
+              type='number'
+              className='input-txt'
+              required
+              value={formState.phoneNumber}
+              onChange={(e) =>
+                formDispatch({ type: "SET_PHONENO", payload: e.target.value })
+              }
+            />
+            <span className='flt-label'>Phone Number</span>
+          </div>
+        </div>
+
+        <div className='flex-input'>
+          <div className='input'>
+            <input
+              type='number'
+              className='input-txt'
+              required
+              value={formState.zipCode}
+              onChange={(e) =>
+                formDispatch({ type: "SET_ZIPCODE", payload: e.target.value })
+              }
+            />
+            <span className='flt-label'>Zip Code</span>
+          </div>
+
+          <div className='input'>
+            <input
+              type='text'
+              className='input-txt'
+              required
+              value={formState.city}
+              onChange={(e) =>
+                formDispatch({ type: "SET_CITY", payload: e.target.value })
+              }
+            />
+            <span className='flt-label'>City</span>
```

Comment:
I cant find delete address button here , but when I delete a address the page goes blank after that action

Reply:
Yes, thank you! This is fixed